daml-assistant/src/DA/Daml/Assistant/Version.hs

```diff
@@ -121,100 +126,151 @@ getDefaultSdkVersion damlPath = do
     required "There are no installed SDK versions." $
         maximumMay installedVersions
 
--- | Get the list of available versions afresh. This will fetch
--- https://docs.daml.com/versions.json and parse the obtained list
--- of versions.
-getAvailableSdkVersions :: IO [SdkVersion]
-getAvailableSdkVersions = wrapErr "Fetching list of available SDK versions" $ do
-    response <- requiredAny "HTTP connection to docs.daml.com failed" $ do
-        request <- parseRequest "GET http://docs.daml.com/versions.json"
-        httpBS request { responseTimeout = responseTimeoutMicro 2000000 }
-
-    when (getResponseStatusCode response /= 200) $ do
-        throwIO $ assistantErrorBecause
-            "Fetching list of available SDK versions from docs.daml.com failed"
-            (pack . show $ getResponseStatus response)
-
-    versionsMap :: M.HashMap Text Text <-
-        fromRightM
-            (throwIO . assistantErrorBecause "Versions list from docs.daml.com does not contain valid JSON" . pack)
-            (eitherDecodeStrict' (getResponseBody response))
-
-    pure . sort $ mapMaybe (eitherToMaybe . parseVersion) (M.keys versionsMap)
-
--- | Get the list of available snapshot versions. This will fetch
--- https://docs.daml.com/snapshots.json and parse the obtained list
--- of versions.
-getAvailableSdkSnapshotVersions :: IO [SdkVersion]
-getAvailableSdkSnapshotVersions = wrapErr "Fetching list of available SDK snapshot versions" $ do
-    response <- requiredAny "HTTP connection to docs.daml.com failed" $ do
-        request <- parseRequest "GET http://docs.daml.com/snapshots.json"
-        httpBS request { responseTimeout = responseTimeoutMicro 2000000 }
-
-    when (getResponseStatusCode response /= 200) $ do
-        throwIO $ assistantErrorBecause
-            "Fetching list of available SDK snapshot versions from docs.daml.com failed"
-            (pack . show $ getResponseStatus response)
-
-    versionsMap :: M.HashMap Text Text <-
+isReleaseVersion :: SdkVersion -> Bool
```

Comment:
Ideally this should ideally also check the `prerelease` attribute of the GitHub API response (it's a stable release if `prerelease` is `false`).

Reply:
fixed